haystack/components/evaluators/statistical_evaluator.py

```diff
@@ -150,3 +152,19 @@ def _recall_multi_hit(labels: List[str], predictions: List[str]) -> float:
                 correct_retrievals += 1
 
         return correct_retrievals / len(labels)
+
+    @staticmethod
+    def _mrr(labels: List[str], predictions: List[str]) -> float:
+        """
+        Measures the mean reciprocal rank of times a label is present in at least one or more predictions.
+        """
+        if len(labels) == 0:
+            return 0.0
+
+        mrr_sum = 0.0
+        for label, prediction in zip(labels, predictions):
+            if label in prediction:
+                mrr_sum += 1 / (predictions.index(prediction) + 1)
+                break
```

Comment:
This doesn't seem correct to me. This way we break as soon as we find the first matching label, we won't check later labels.

Shouldn't the logic be equivalent to this?
```
for label in labels:
    for rank, prediction in enumerate(predictions):
        if label in prediction:
            mrr_sum += 1 / (rank + 1)
            break
```

Given the data used in `TestStatisticalEvaluatorMRR::test_run` I also get the correct result of `0.3333333333333333`. Calculating by hand it's also correct `(1+1/3+0+0)/4 == 0.333333`.

Reply:
You're right, it does not check beyond the first matching label. I will update the logic to fix that.